src/internal/base-component/get-data-attributes.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export function getDataAttributes(props: Record<string, string>) {
+  const result: Record<string, any> = {};
```

Comment:
It is not always a string, `data-something={true}` is also allowed.

Since we do not convert user input in any way, just pass as is, `any` is fine.

Also, this code is an adapted version of from the main components repo (removed `id` and `className` support): https://github.com/cloudscape-design/components/blob/574a45b2be3c60702be73db8939622af269e6b0e/src/internal/base-component/index.ts#L28-L36